projects/HVAC/vent_control/vent.ino

```diff
@@ -0,0 +1,260 @@
+const char* net_address = "VENT01";
+
+
+void setup() {
+  // Serial
+  Serial.begin(9600);
+
+  // Error light
+  pinMode(blink_pin, OUTPUT);
+  digitalWrite(blink_pin, LOW);
+
+  // Motor pins
+  pinMode(8,  OUTPUT);
+  pinMode(9,  OUTPUT);
+  pinMode(10, OUTPUT);
+  pinMode(11, OUTPUT);
+
+  pinMode(motor_sensor_pin, INPUT);
+
+  if(!motor_zero()) {
+    set_error("Cannot zero motor!");
+  }
+
+  strcpy(net.deviceID, net_address);
+
+  blink_happy();
+  net.sendResponse("READY");
+
+}
+
+void loop() {
+  // Example motor handling
+  // delay(1000);
+  // motor_rotate_to(128);
+  // motor_rotate_degrees(90);
+  // motor_rotate_percents(25);
+  // delay(1000);
+  // motor_rotate_to(0);
+  // motor_rotate_degrees(0);
+  // motor_rotate_percents(0);
+  // delay(1000);
+
+  if(net.messageReceived()) {
+    process_command();
+  }
+}
+
+// blink successfull init
+void blink_happy() {
+  for(int i = 0; i < 5; i++) {
+    digitalWrite(blink_pin, HIGH);
+    delay(100);
+    digitalWrite(blink_pin, LOW);
+    delay(50);
+  }
+}
+
+// -----------
+
+void process_command() {
+  char buffer [100];
+  long int value = 0;
+
+  if(net.assertCommandStarts("set degrees:", buffer)) {
+    value = strtol(buffer, NULL, 10);
+    motor_rotate_degrees(value);
+    net.sendResponse("OK");
+    return;
+  }
+
+  if(net.assertCommandStarts("set percents:", buffer)) {
+    value = strtol(buffer, NULL, 10);
+    motor_rotate_percents(value);
+    net.sendResponse("OK");
+    return;
+  }
+
+  net.sendResponse("ERROR");
+}
+
+// ----------- MOTOR ROUTINES -------------
+
+void motor_step_cw() {
+  step1();
+  step2();
+  step3();
+  step4();
+  step5();
+  step6();
+  step7();
+  step8();
+}
+
+void motor_step_ccw() {
+  step8();
+  step7();
+  step6();
+  step5();
+  step4();
+  step3();
+  step2();
+  step1();
+}
+
+void step1(){
+  digitalWrite(8, HIGH);
+  digitalWrite(9, LOW);
+  digitalWrite(10, LOW);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step2(){
+  digitalWrite(8, HIGH);
+  digitalWrite(9, HIGH);
+  digitalWrite(10, LOW);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step3(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, HIGH);
+  digitalWrite(10, LOW);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step4(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, HIGH);
+  digitalWrite(10, HIGH);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step5(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, LOW);
+  digitalWrite(10, HIGH);
+  digitalWrite(11, LOW);
+  delay(2);
+}
+void step6(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, LOW);
+  digitalWrite(10, HIGH);
+  digitalWrite(11, HIGH);
+  delay(2);
+}
+void step7(){
+  digitalWrite(8, LOW);
+  digitalWrite(9, LOW);
+  digitalWrite(10, LOW);
+  digitalWrite(11, HIGH);
+  delay(2);
+}
+void step8(){
+  digitalWrite(8, HIGH);
+  digitalWrite(9, LOW);
+  digitalWrite(10, LOW);
+  digitalWrite(11, HIGH);
+  delay(2);
+}
+
+bool motor_zero() {
+  for(int i = 0; i < motor_steps_per_rotation; i++) {
+    motor_step_cw();
```

Comment:
It is likely that depending which way we approach the sensor, damper will end up in different position. E.g. if we go clockwise, it will be 10% open when switch is triggered, and 15% open if we co CCW. I'd suggest moving this to configuration.